### PR TITLE
Site Settings: Introduce SSO card and wrapped security form component

### PR DESCRIPTION
**Pull request by tyxla**
This PR introduces the SSO card in the Security site settings section. It also introduces a wrapped security form component that we will use for adding the new modern cards into the Security settings section. 

Preview of the new SSO card:

A full list of what this PR suggests is:

* Introducing a new SSO settings card component.
* Introducing a new wrapped security form settings component.
* Embedding the new security form component into the security form section.
* Including the SSO settings in the `filterSettingsByActiveModules` utility and its tests.
* Introducing a new CSS class for security settings wrapper.

To test:

* Checkout this branch
* Go to `/settings/security/$site` where `$site` is a Jetpack site.
* Verify the SSO card looks as on the screenshot.
* Play with the main toggle and the sub settings toggles, verify they work properly when retrieving and storing data, comparing with the values in your wp-admin.
* Try saving with disabled main toggle and verify it still saves correctly.
* Test with dev mode and verify the main toggle and sub toggles are properly disabled.
* Verify all util `filterSettingsByActiveModules` tests still pass:

```
npm run test-client client/state/jetpack/settings/test/utils.js -- --grep="filterSettingsByActiveModules"
```

### CODE REVIEW

File: client/my-sites/site-settings/form-security.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+		return (
+			<form
+				id="site-settings"
+				onSubmit={ handleSubmitForm }
+				className="site-settings__security-settings"
+			>
+				<QueryJetpackModules siteId={ siteId } />
+
+				{ this.renderSectionHeader( translate( 'WordPress.com log in' ) ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-security.jsx]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Log in to WordPress.com' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/sso.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+						>
+							{ translate( 'Match accounts using email addresses' ) }
+						</CompactFormToggle>
+
+						<CompactFormToggle
+							checked={ !! fields.jetpack_sso_require_two_step }
+							disabled={ isRequestingSettings || isSavingSettings || ! ssoModuleActive }
+							onChange={ handleToggle( 'jetpack_sso_require_two_step' ) }
+						>
+							{ translate( 'Require two step authentication' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/sso.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Activate Two Step Authentication' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/sso.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+							<ExternalLink href={ 'https://jetpack.com/support/sso' } target="_blank">
+								{ translate( 'Learn more about WordPress.com Secure Sign On' ) }
+							</ExternalLink>
+						</InfoPopover>
+					</div>
+
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="sso"
+						label={ translate( 'Allow log-in using WordPress.com accounts.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/sso.jsx]: Hi! I've found a possible matching string that has already been translated **11** times:
`translate( 'Allow your users to log in using their WordPress.com accounts.' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by oskosk** [comment]
Looking good so far!. Can we add the disabling of these settings when in dev mode to this PR

**Comment by tyxla** [comment]
Thanks for the review @osk!

> Looking good so far!. Can we add the disabling of these settings when in dev mode to this PR?

Ah, good call. Done in https://github.com/Automattic/wp-calypso/pull/11881/commits/2bba1055f61c928c6939ad8b48d2f0073aa44513.

> What if we add auto-saving toggles to this PR too?

Cool idea! Done in https://github.com/Automattic/wp-calypso/pull/11881/commits/f6d5c869937901e04d9fa43057edb023df15a21f.

Feel free to have another 👁  @oskosk

File: client/my-sites/site-settings/form-security.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+		return (
+			<form
+				id="site-settings"
+				onSubmit={ handleSubmitForm }
+				className="site-settings__security-settings"
+			>
+				<QueryJetpackModules siteId={ siteId } />
+
+				{ this.renderSectionHeader( translate( 'WordPress.com log in' ), true, ! ssoModuleActive || ssoModuleUnavailable ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-security.jsx]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Log in to WordPress.com' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/sso.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+						>
+							{ translate( 'Match accounts using email addresses' ) }
+						</CompactFormToggle>
+
+						<CompactFormToggle
+							checked={ !! fields.jetpack_sso_require_two_step }
+							disabled={ isRequestingSettings || isSavingSettings || ! ssoModuleActive || ssoModuleUnavailable }
+							onChange={ handleAutosavingToggle( 'jetpack_sso_require_two_step' ) }
+						>
+							{ translate( 'Require two step authentication' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/sso.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Activate Two Step Authentication' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/form-security.jsx (hunk 1 above)
Comment (oskosk) [inline client/my-sites/site-settings/form-security.jsx]: I have my doubts here about if we should put `WordPress.com login` (`login` being a noun) or leave it as `WordPress.com log in` (`log in` is verb)

Comment (tyxla) [inline client/my-sites/site-settings/form-security.jsx]: I have no preferences on this one, really. I've just followed what's on p6TEKc-HQ-p2 - @rickybanister @MichaelArestad what do you think we should be using here?

File: client/my-sites/site-settings/form-security.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+		return (
+			<form
+				id="site-settings"
+				onSubmit={ handleSubmitForm }
+				className="site-settings__security-settings"
+			>
+				<QueryJetpackModules siteId={ siteId } />
+
+				{ this.renderSectionHeader( translate( 'WordPress.com log in' ), false ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-security.jsx]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Log in to WordPress.com' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
Looks good, there are a few copy-tweaks however.

Let's make sure it says 'WordPress.com sign in' two words, and avoid the word 'log'.

**Comment by oskosk** [comment]
LGTM

**Comment by tyxla** [comment]
@rickybanister I've updated the copy as you suggested, here is an updated preview:



Anything else design-wise?

**Comment by rickybanister** [comment]
Can we remove the period at the end of the toggle's label. And two-step should have a hyphen. 

Otherwise looks good.

**Comment by tyxla** [comment]
Done:



Thanks for the review!

File: client/my-sites/site-settings/form-security.jsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+		return (
+			<form
+				id="site-settings"
+				onSubmit={ handleSubmitForm }
+				className="site-settings__security-settings"
+			>
+				<QueryJetpackModules siteId={ siteId } />
+
+				{ this.renderSectionHeader( translate( 'WordPress.com sign in' ), false ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-security.jsx]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Sign in to WordPress.com' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/sso.jsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+						>
+							{ translate( 'Match accounts using email addresses' ) }
+						</CompactFormToggle>
+
+						<CompactFormToggle
+							checked={ !! fields.jetpack_sso_require_two_step }
+							disabled={ isRequestingSettings || isSavingSettings || ! ssoModuleActive || ssoModuleUnavailable }
+							onChange={ handleAutosavingToggle( 'jetpack_sso_require_two_step' ) }
+						>
+							{ translate( 'Require two-step authentication' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/sso.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Activate Two Step Authentication' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-sso-card)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**